<img src="http://imgur.com/1ZcRyrc.png" style="float: left; margin: 20px; height: 55px">

# Bayes' Rule Practice Lab

_Author: Matt Brems_

This lab will have two parts:

- Part 1: Bayes Practice Problems
- Part 2: Moving from Probabilities to Distributions

---

## Part 1: Bayes Practice Problems

### Problem 1: IRAs

You work at a financial services company that offers individual retirement accounts (IRAs). To target high potential customers, you want to find out if people with children are more likely to invest in IRAs.  

So far you have found:
- 30% of all Americans have IRAs.
- half of Americans have children.
- two-thirds of those with IRAs have children.

Based on this information, what is the probability that someone with children has an IRA? 

In [2]:
p_chil_ira = 2/3
p_chil = .5
p_ira = .3
p_ira_chil = (p_chil_ira*p_ira)/p_chil
p_ira_chil

0.39999999999999997

**Answer: 40%** 

---

### Problem 2: Resident Satisfaction

Suppose I own a hotel and an apartment building. Times are tight, and I need to sell one of them. Either would earn me the exact same amount of money, so that doesn't help me to make my decision. Instead, I will make my decision based on my residents' satisfaction. 

- Whichever building has a higher percentage of satisfied people is the one I will choose to keep.
- *However*, if the percent of satisfied guests for my hotel and the percent of satified guests for my apartment building are within 10% of one another, I will sell neither until I can bring in some outside consultants to help me make my decision.
- Last winter, I administered a survey to *all* residents and found that 60% of them are satisfied.
- Among satisfied respondents, two out of every five came from the apartment building.
- One-third of my residents live in my apartment building.

What action will I take?

In [1]:
p_satisfied = .6
p_satisfied_apart = 2/5
p_apart = 1/3
p_hotel = 2/3
p_satisfied_hotel = 3/5
p_satisfied_apart - p_satisfied_hotel


-0.19999999999999996

**Answer:** Apartment residents are less satisfied so you should sell the apartment building

---

### Problem 3: Coin Game

Suppose you and your friend are playing a game. Your friend has laid four coins out in front of you. If you flip heads, you win a dollar from your friend. If you flip tails, you owe a dollar to your friend. However, the coins in front of you are not fair.
- One coin has a 80% chance of flipping heads. (Call this coin A.)
- One coin has a 60% chance of flipping heads. (Call this coin B.)
- One coin has a 40% chance of flipping heads. (Call this coin C.)
- One coin has a 10% chance of flipping heads. (Call this coin D.)

**Problem 3 (a)**: Suppose you select one coin at random. That is, you don't know whether you selected coin A, B, C, or D. You flip heads. Given this data, what are the probabilities that you selected coin A, coin B, coin C, and coin D?

> Hint: When finding the marginal probability of flipping heads, you will need to use the Law of Total Probability! Find this **first**.

In [5]:
p_heads_a = .8
p_heads_b = .6
p_heads_c = .4
p_heads_d = .1
p_a, p_b, p_c, p_d = .25, .25, .25, .25
p_heads = p_heads_a*p_a+p_heads_b*p_b+p_heads_c*p_c+p_heads_d*p_d
p_heads

0.475

In [7]:
p_a_heads = p_heads_a*p_a/p_heads
p_a_heads

0.4210526315789474

In [8]:
p_b_heads = p_heads_b*p_b/p_heads
p_b_heads

0.3157894736842105

In [9]:
p_c_heads = p_heads_c*p_c/p_heads
p_c_heads

0.2105263157894737

In [10]:
p_d_heads = p_heads_d*p_d/p_heads
p_d_heads

0.052631578947368425

**Answer**: Respectively for a, b, c and d: .4211, .3158, .2105, .0526

**BONUS: Problem 3 (b)**: Suppose you are using the same coin as before. That is, you _still_ don't know whether you selected coin A, B, C, or D - although you have a better idea now that you flipped heads on your first flip! On this second flip, you flip tails. Given this data, what are the probabilities that you selected coin A, coin B, coin C, and coin D?

> Hint: Everything now is **conditional on the fact that we have already flipped a heads**.

$$
\begin{eqnarray*}
P(A|B\cap C) &=& \frac{P(A\cap B \cap C)}{P(B\cap C)} \\
\\
&=& \frac{P(A\cap B|C)P(C)}{P(B|C)P(C)} \\
\\
&=& \frac{P(A\cap B|C)}{P(B|C)} \\
\end{eqnarray*}
$$

**Answer**:

**BONUS: Problem 3(c)**: Suppose you are using the same coin as before. That is, you _still_ don't know whether you selected coin A, B, C, or D - although you have a better idea now that you flipped heads, then tails on your first two flips! On this third flip, you flip tails. Given this data, what are the probabilities that you selected coin A, coin B, coin C, and coin D?

> Hint: This will follow the same structure as the last two problems.

**Answer**:

---

## Part 2: Moving from Probabilities to Distributions

### Problem 4: Prior Probabilities

**Problem 4 (a)**: In Problem 3, before you had flipped any coin, what was the probability that you had selected coin A? Coin B? Coin C? Coin D?

**Answer:** .25

**Problem 4 (b)**: What is the definition of a distribution?
> Check your notes from the early weeks of the course if you can't remember.

**Answer:** The function that gives the probabilities of all possible events in an experiment


**Problem 4 (c)**: What "named distribution" (i.e. a distribution that has a common name) could we apply to a situation where selecting coins A-D is equally likely?

**Answer:** Discrete uniform distribution

**Problem 4 (d)**: Suppose that I have only one coin and I want to conduct inference on the probability of flipping heads, $p$. Note that $p$ is unknown. If I want to specify some prior distribution where all probabilities $p$ are equally likely, should I use a discrete or continuous distribution? Why?

**Answer:** A discrete distribution because all outcomes are discrete and equally likely

**Problem 4 (e)**: Suppose that I have only one coin and I want to conduct inference on the probability of flipping heads, $p$. Note that $p$ is unknown. If I want to specify some prior distribution where all probabilities $p$ are equally likely, then what named distribution might I use?

**Answer:** Discrete uniform distribution

**Problem 4 (f)**: Suppose I have only one coin with some unknown probability of flipping heads $p$. If I think 50% is the likeliest value for $p$ and, as we get farther away from 50%, that value is less and less likely, what named distribution could I use?

**Answer:** Normal distribution

**Problem 4 (g)**: When we listed out our prior probabilities for each coin, we got the same number of posterior probabilities. For example, when we had four prior probabilities (one for Coin A, one for Coin B, one for Coin C, and one for Coin D), then we had four posterior probabilities (one for Coin A, for Coin B, one for Coin C, and one for Coin D). However, if we have one prior distribution, then we will get one posterior distribution. What are some advantages to being able to summarize our posterior with one distribution instead of a big list of probabilities?

**Answer:**